.circleci/config.yml

```diff
@@ -80,140 +80,140 @@ workflows:
-        parallelism: 4
-        steps:
-            - checkout
-            - run:
-                  name: Install python dependencies
-                  command: |
-                      pip3 install tox==3.24.0
-            - restore-tox-cache
-            - run:
-                  name: Run tests
-                  # Tests failed with Too many open files, so added ulimit
-                  command: |
-                      ulimit -n 4096
-                      CI_PYTEST_PARALLEL=3 CI_PYTEST_SPLIT_ARGS="--splits $CIRCLE_NODE_TOTAL --group $(( $CIRCLE_NODE_INDEX + 1 ))" python3 -m tox -v -e << parameters.toxenv >>
-                  no_output_timeout: 10m
-            - save-tox-cache
-            - save-test-results
+  test:
+    resource_class: xlarge
+    parameters:
+      image:
+        type: string
+      toxenv:
+        type: string
+    docker:
+      - image: << parameters.image >>
+    working_directory: /mnt/ramdisk
+    steps:
+      - checkout
+      - run:
+          name: Install system deps
+          command: apt-get update && apt-get install -y libsndfile1 ffmpeg
+      - run:
+          name: Install python dependencies
+          command: |
+            pip install tox==3.24.0
+      - restore-tox-cache
+      - run:
+          name: Run tests
+          command: |
+            tox -v -e << parameters.toxenv >>
+          no_output_timeout: 10m
+      - save-tox-cache
+      - save-test-results
+  win:
+    parameters:
+      toxenv:
+        type: string
+    executor: win/default
+    parallelism: 4
+    steps:
+      - checkout
+      - run:
+          name: Install python dependencies
+          command: |
+            pip install tox==3.24.0
+      - restore-tox-cache
+      - run:
+          name: Temporary conda hack
+          shell: bash.exe
+          command: |
+            cp /c/tools/miniconda3/python* /c/tools/miniconda3/lib/venv/scripts/nt/
+      - run:
+          name: Run tests
+          shell: bash.exe
+          command: |
+            CI_PYTEST_PARALLEL=3 CI_PYTEST_SPLIT_ARGS="--splits $CIRCLE_NODE_TOTAL --group $(( $CIRCLE_NODE_INDEX + 1 ))" tox -v -e << parameters.toxenv >>
+          no_output_timeout: 10m
+      - save-tox-cache
+      - save-test-results
+  mac:
+    # TODO: how to set resource class?
+    parameters:
+      toxenv:
+        type: string
+    macos:
+      xcode: 11.4.1
+    parallelism: 4
+    steps:
+      - checkout
+      - run:
+          name: Install python dependencies
+          command: |
+            pip3 install tox==3.24.0
+      - restore-tox-cache
+      - run:
+          name: Run tests
+          # Tests failed with Too many open files, so added ulimit
+          command: |
+            ulimit -n 4096
+            CI_PYTEST_PARALLEL=3 CI_PYTEST_SPLIT_ARGS="--splits $CIRCLE_NODE_TOTAL --group $(( $CIRCLE_NODE_INDEX + 1 ))" python3 -m tox -v -e << parameters.toxenv >>
+          no_output_timeout: 10m
+      - save-tox-cache
+      - save-test-results
```

Comment:
These formatting changes are unfortunate.  Not a deal breaker but would be cleaner to just remove all changes to this file.  It looks like master is using the 0.22 keys anyway so there are actually no changes in this file.

Reply:
please revert this file.

Reply:
done